### ISSUE TITLE
NTIRE2018_Validate data

### ISSUE THREAD

**Issue by yumengying5** [abse4411/hscnn-plus#1]
Hello, I don't have the NTIRE2018_Validate data, and I can't download data set from official website, but I need it badly. could you share it ? Thank you very much！

**Comment by abse4411** [comment]
Can you provide your E-mail address? I will send you with the NTIRE2018_Validate data later.

**Comment by yumengying5** [comment]
E-mail address: yumengying5@163.com
thank you again.

**Comment by abse4411** [comment]
An E-mail with data was sent, check your inbox.

**Comment by Dave-SUN** [comment]
Hello, when I run generate_data_rgb_ntire.m, matlab is always unresponsive, I don't know what's wrong with me, I hope you can give me some guidance, thank you very much. What I did was to create the folder "data" and create the folders "train_data" and "valid" in it respectively. In the folder "train_data", I created the folders "rgb_real" and "rgb_clean" respectively. In the folder "valid" create the folders "validClean" and "validReal" respectively. I'm confused, I don't know what to put in the above folders, I hope you can help me, thank you very much.

**Comment by abse4411** [comment]
Hello, @Dave-SUN . Thanks for your interest in my early work. The file "generate_data_rgb_ntire.m" is to generate train data and valid one (store all hs data and corresponding rgb images in a hdf5 file) for hscnn implemented by authors. I rewrote dataset file which read data in another manner. As a result, the matlab script file are no longer used.
The folder you created is correct. A complete structure of folders is as follows:
================ The Structure of data ================ 
--data
    --train_data
        --mat
            -- BGU_HS_00001.mat
            -- ...(mat files of NTIRE2018 clean track or realworld track)
        --rgb_clean
            --BGU_0522-1217_clean.png
            -- ...(rgb files of NTIRE2018 clean track)
        --rgb_real
            --BGU_0522-1217_camera.jpg
            -- ...(rgb files of NTIRE2018 clean track)
    --valid(similar structure to train_data, but for valid data)
        --mat
            -- BGU_HS_00257.mat
            -- ...(valid mat files of NTIRE2018 clean track or realworld track)
        --validClean
            --BGU_HS_00257_clean.png
            -- ...(valid  rgb files of NTIRE2018 clean track)
        --validReal
            --BGU_HS_00257_camera.jpg
            -- ...(valid rgb files of NTIRE2018 clean track)
================ End ================ 
I uploaded the original code for hscnn, you can download it from following link:
link：https://pan.baidu.com/s/17WYP7VG7m3arMz62WPagfQ?pwd=gi0d 
access code：gi0d

**Comment by Dave-SUN** [comment]
Thank you very much for your reply, you mentioned that you rewrote dataset file which read data in another manner, I wonder if you can share this new version? Also, the official URL of NTIRE2018_Validate data is no longer available, could you please provide me with a copy? Thank you very much!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "abse4411/hscnn-plus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月23日(星期三) 中午11:43
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [abse4411/hscnn-plus] NTIRE2018_Validate data (#1)





 
Hello, @Dave-SUN . Thanks for your interest in my early work. The file "generate_data_rgb_ntire.m" is to generate train data and valid one (store all hs data and corresponding rgb images in a hdf5 file) for hscnn implemented by original authors. I rewrote dataset file which read data in another manner. As a result, the matlab script file are no longer used.
 The folder you created is correct. A complete structure of folders is as follows:
 ================ The Structure of data ================
 --data
 --train_data
 --mat
 -- BGU_HS_00001.mat
 -- ...(mat files of NTIRE2018 clean track or realworld track)
 --rgb_clean
 --BGU_0522-1217_clean.png
 -- ...(rgb files of NTIRE2018 clean track)
 --rgb_real
 --BGU_0522-1217_camera.jpg
 -- ...(rgb files of NTIRE2018 clean track)
 --valid(similar structure to train_data, but for valid data)
 --mat
 -- BGU_HS_00257.mat
 -- ...(valid mat files of NTIRE2018 clean track or realworld track)
 --validClean
 --BGU_HS_00257_clean.png
 -- ...(valid  rgb files of NTIRE2018 clean track)
 --validReal
 --BGU_HS_00257_camera.jpg
 -- ...(valid rgb files of NTIRE2018 clean track)
 ================ End ================
 I uploaded the original code for hscnn, you can download it from following link:
 link：https://pan.baidu.com/s/17WYP7VG7m3arMz62WPagfQ?pwd=gi0d
 access code：gi0d

**Comment by abse4411** [comment]
Hi @Dave-SUN , thanks for your interest in my early work. 
1. The dataset file what I mentioned is in current repository, see it in "train/dataset.py". Please note that the dataset class is not designed for reading NTIRE2018 data. Actually, it reads data from folders where each RGB image and single-channel images(gray images for each channel of the HSI) are put together, for example(CAVE):
--CAVE
----balloons_ms
--------balloons_ms_RGB.bmp
--------balloons_ms_01.png
--------balloons_ms_01.png
--------balloons_ms_02.png
--------...

2. I updated readme.md file in this repository. You can find out the link to download it.

**Comment by Dave-SUN** [comment]
Thank you very much for your reply, I will study your instructions carefully




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "abse4411/hscnn-plus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月23日(星期三) 中午1:58
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [abse4411/hscnn-plus] NTIRE2018_Validate data (#1)





 
Hi @Dave-SUN , thanks for your interest in my early work.
  
 
The dataset file what I mentioned is in current repository, see it in "train/dataset.py". Please note that the dataset class is not designed for reading NTIRE2018 data. Actually, it reads data from folders where each RGB image and single-channel images(gray images for each channel of the HSI) are put together, for example(CAVE):
 --CAVE
 ----balloons_ms
 --------balloons_ms_RGB.bmp
 --------balloons_ms_01.png
 --------balloons_ms_01.png
 --------balloons_ms_02.png
 --------...
 
 
 
I updated readme.md file in this repository. You can find out the link to download it.

**Comment by Dave-SUN** [comment]
Hi, according to the code of hscnn_main and generate_data_rgb_ntire.m files, I can tell that the final train_data and val_data sent to the network are both .h5 format files. What puzzles me is that the h5 files generated by my matlab only have the training set starting with "train" but not the validation set starting with "clean". Thank you for your patience.






------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "LOK"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月23日(星期三) 下午2:03
***@***.***&gt;;

主题:&nbsp;回复： [abse4411/hscnn-plus] NTIRE2018_Validate data (#1)



Thank you very much for your reply, I will study your instructions carefully




------------------ 原始邮件 ------------------
发件人:                                                                                                                        "abse4411/hscnn-plus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月23日(星期三) 中午1:58
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [abse4411/hscnn-plus] NTIRE2018_Validate data (#1)





 
Hi @Dave-SUN , thanks for your interest in my early work.
  
 
The dataset file what I mentioned is in current repository, see it in "train/dataset.py". Please note that the dataset class is not designed for reading NTIRE2018 data. Actually, it reads data from folders where each RGB image and single-channel images(gray images for each channel of the HSI) are put together, for example(CAVE):
 --CAVE
 ----balloons_ms
 --------balloons_ms_RGB.bmp
 --------balloons_ms_01.png
 --------balloons_ms_01.png
 --------balloons_ms_02.png
 --------...
 
 
 
I updated readme.md file in this repository. You can find out the link to download it.

**Comment by abse4411** [comment]
Hi @Dave-SUN , thanks for your interest in my early work.

In generate_data_rgb_ntire.m file, there is a variable named "string" which determinate where to read mat files. You should change its value (e.g. val) rather than the default value 'train'.
This is partial content of file "generate_data_rgb_ntire.m"
===========================
`string = 'train';
string2 = 'clean';
if strcmp(string, 'train') == 1

    hyper_dir = '../../data/train_data/mat/';
    label=dir(fullfile(hyper_dir,'*.mat'));

    
    if strcmp(string2, 'clean') == 1   
        rgb_dir = '../../data/train_data/rgb_clean/';
    else  
        rgb_dir = '../../data/train_data/rgb_real/';    
    end
    
     order= randperm(size(label,1));
    
    
else

    hyper_dir = '../../data/valid/mat/';
    label=dir(fullfile(hyper_dir,'*.mat'));
   
    
    if strcmp(string2, 'clean') == 1   

        rgb_dir = '../../data/valid/validClean/';    
    else  
         rgb_dir = '../../data/valid/validReal/';
 
    end
    
     order= randperm(size(label,1));
    
end`
===========================

**Comment by Dave-SUN** [comment]
Thank you very much.



发自我的iPhone


------------------ Original ------------------
From: abse4411 ***@***.***&gt;
Date: Wed,Mar 23,2022 5:01 PM
To: abse4411/hscnn-plus ***@***.***&gt;
Cc: Dave-SUN ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [abse4411/hscnn-plus] NTIRE2018_Validate data (#1)





 
Hi @Dave-SUN , thanks for your interest in my early work.
 
In generate_data_rgb_ntire.m file, there is a variable named "string" which determinate where to read mat files. You should change its value (e.g. val) rather than the default value 'train'.
 This is partial content of file "generate_data_rgb_ntire.m"
 
`string = 'train';
 string2 = 'clean';
 if strcmp(string, 'train') == 1
 hyper_dir = '../../data/train_data/mat/'; label=dir(fullfile(hyper_dir,'*.mat')); if strcmp(string2, 'clean') == 1        rgb_dir = '../../data/train_data/rgb_clean/'; else       rgb_dir = '../../data/train_data/rgb_real/';     end  order= randperm(size(label,1));  
else
 hyper_dir = '../../data/valid/mat/'; label=dir(fullfile(hyper_dir,'*.mat')); if strcmp(string2, 'clean') == 1        rgb_dir = '../../data/valid/validClean/';     else        rgb_dir = '../../data/valid/validReal/'; end  order= randperm(size(label,1));  
end`

**Comment by Dave-SUN** [comment]
I successfully ran hscnn_main.py in the train folder and test_model1.py, test_model2.py，test_model3.py and ensemble.py in the test folder，but the results in the final_results folder are in .mat format as shown in the figure. I would like to know how to get the visualization results as mentioned in the paper and the quality analysis evaluation? Thank you for your patience!



------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "LOK"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月23日(星期三) 下午4:31
***@***.***&gt;;

主题:&nbsp;回复： [abse4411/hscnn-plus] NTIRE2018_Validate data (#1)



Hi, according to the code of hscnn_main and generate_data_rgb_ntire.m files, I can tell that the final train_data and val_data sent to the network are both .h5 format files. What puzzles me is that the h5 files generated by my matlab only have the training set starting with "train" but not the validation set starting with "learn". Thank you for your patience.






------------------ 原始邮件 ------------------
发件人:                                                                                                                        "LOK"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月23日(星期三) 下午2:03
***@***.***&gt;;

主题:&nbsp;回复： [abse4411/hscnn-plus] NTIRE2018_Validate data (#1)



Thank you very much for your reply, I will study your instructions carefully




------------------ 原始邮件 ------------------
发件人:                                                                                                                        "abse4411/hscnn-plus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年3月23日(星期三) 中午1:58
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [abse4411/hscnn-plus] NTIRE2018_Validate data (#1)





 
Hi @Dave-SUN , thanks for your interest in my early work.
  
 
The dataset file what I mentioned is in current repository, see it in "train/dataset.py". Please note that the dataset class is not designed for reading NTIRE2018 data. Actually, it reads data from folders where each RGB image and single-channel images(gray images for each channel of the HSI) are put together, for example(CAVE):
 --CAVE
 ----balloons_ms
 --------balloons_ms_RGB.bmp
 --------balloons_ms_01.png
 --------balloons_ms_01.png
 --------balloons_ms_02.png
 --------...
 
 
 
I updated readme.md file in this repository. You can find out the link to download it.

**Comment by abse4411** [comment]
Hi @Dave-SUN , thanks for your interest in my early work.
The code for results of visualization and evaluation is not provided in original one. You should implement on your own, or look for other codes which work on spectral super-resolution to get corresponding implementation details. Thanks! :-)

**Comment by Maggie84** [comment]
hello, I can't download the NTIRE2018_Validate data from official website. But I need it badly. Could you share NTIRE2018 data set? 
If you have NTIRE2018_Test data, very sincerely hope can send me together.
My E-mail address: 744316958@qq.com. Thank you very much.

**Comment by abse4411** [comment]
> hello, I can't download the NTIRE2018_Validate data from official website. But I need it badly. Could you share NTIRE2018 data set? If you have NTIRE2018_Test data, very sincerely hope can send me together. My E-mail address: [744316958@qq.com](mailto:744316958@qq.com). Thank you very much.

Sorry, the official website does not provide the test data. Therefore, I don't have it either.

**Comment by Maggie84** [comment]
thank you very much!




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "abse4411/hscnn-plus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年5月16日(星期一) 晚上7:06
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [abse4411/hscnn-plus] NTIRE2018_Validate data (#1)





  
hello, I can't download the NTIRE2018_Validate data from official website. But I need it badly. Could you share NTIRE2018 data set? If you have NTIRE2018_Test data, very sincerely hope can send me together. My E-mail address: ***@***.*** Thank you very much.
  
Please visit this website https://pan.baidu.com/s/1KQlPTLQRQ4R5bc67AqRYhA?pwd=3esw to download it.
